### ISSUE TITLE
Customize validator returns library internal error

### ISSUE THREAD

**Issue by tranvansang** [express-validator/express-validator#593]
Reproduction:

`package.json`
```
{
  "name": "w",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "body-parser": "^1.18.3",
    "express": "^4.16.3",
    "express-validator": "^5.2.0"
  }
}
```

`app.json`
```javascript
const {body, validationResult} = require('express-validator/check')
require('express')()
  .use(require('body-parser').json())
  .use(
    body('pwd').isLength({min: 8}, (pwd, {req}) => 'length must >= 8'),
    body('pwdCfrm').custom((pwdCfrm, {req}) => {
      if (pwdCfrm !== req.body.pwd)
        throw new Error('pwds do not match')
    }),
    (req, res) =>  res.end(JSON.stringify(validationResult(req).array()))
  ).listen(8000, () => console.log('app is listening'))
```

result when opening 'localhost:8000' in browser
```
[{"location":"body","param":"pwd","msg":"Invalid value"},{"location":"body","param":"pwdCfrm","msg":"Cannot read property 'then' of undefined"}]
```

while expected result is only `'length must >= 8'` error

I also tried some workaround but thing does not work at all

**Comment by tranvansang** [comment]
it looks like, if `custom` function return `Promise.resolve()`, thing will work

I think there may be somewhere when executing custom validation function, you forget to wrap the result by `Promise.resolve()`

**Comment by gustavohenke** [comment]
Hello @tranvansang, thanks for reporting this bug with a reproducible code!

If you issue a request without passing `pwd` or `pwdCfrm`, then both will be `undefined` in the eyes of express-validator, what means `pwdCfrm` is _actually correct_... so your custom validator doesn't throw.
And this is where the bug lies: we're not ready to deal with validators that return nothing.

Looking at fixing it!

**Comment by gustavohenke** [comment]
Shipped on [v5.3.0](https://github.com/express-validator/express-validator/releases/tag/v5.3.0)! 🚢

**Comment by xdevnull** [comment]
It still occur.

```
    body('domain')
                .matches(/^@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/gi)
                .withMessage('Invalid domain')
                .custom(value => {
                    if(DOMAINS.indexOf(value) == -1) {
                        throw new Error("blah blah blah");
                    }
                }),
```
